version: '3.8'

services:
  openvpn-client:
    image: kylemanna/openvpn
    container_name: openvpn-client
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun
    volumes:
      - C:/Users/OkanBÖKE/Desktop/docker_python_vpn/vpn/dockvpn.ovpn:/etc/openvpn/config.ovpn:ro
      - C:/Users/OkanBÖKE/Desktop/docker_python_vpn/vpn/auth.txt:/etc/openvpn/auth.txt:ro
    command: >
      sh -c "openvpn --config /etc/openvpn/config.ovpn --auth-user-pass /etc/openvpn/auth.txt"
    restart: unless-stopped
    networks:
      - vpn-net

  python-proxy:
    build: ./proxy
    container_name: python-proxy
    ports:
      - "5000:5000"
    networks:
      - vpn-net
    depends_on:
      - openvpn-client

networks:
  vpn-net:
    driver: bridge
